validation.go

```diff
@@ -18,9 +18,13 @@ func init() {
 	validation.AddRule("join", &validation.RuleDefinition{
 		Function: validateJoin,
 	})
+	validation.AddRule("search", &validation.RuleDefinition{
+		Function: validateSearch,
+	})
 	lang.SetDefaultValidationRule("filter.array", "The filter format is invalid.")
 	lang.SetDefaultValidationRule("join.array", "The join format is invalid.")
 	lang.SetDefaultValidationRule("sort.array", "The sort format is invalid.")
+	lang.SetDefaultValidationRule("search.string", "The search format is invalid.")
```

Comment:
```suggestion
	lang.SetDefaultValidationRule("search", "The search format is invalid.")
```
This rule is not type-dependent so the ".string" part is not necessary.